### PR TITLE
Stabilize the panic_col feature

### LINKED ISSUE #42939: Tracking issue for the `panic_col` library feature
Tracking issue for the `panic_col` lib feature added for the column info API in the panic location hook mechanism, added in PR #42938.

### PR DESCRIPTION
**Pull request by est31**
I've added the panic_col feature in PR #42938.
Now it's time to stabilize it!
Closes #42939.

### CODE REVIEW

**Comment by TimNN** [comment]
cc @rust-lang/libs: There seems nothing more to do here, does this require any kind of FCP?

**Comment by alexcrichton** [comment]
@rfcbot fcp merge

Yeah for stabilizations we do indeed like to go through FCP!

**Comment by dtolnay** [comment]
Is it intentional that line numbers in a panic start at 1 but column numbers start at 0? I see that this is what `line!()` and `column!()` do as well. What are the advantages? Is it too late to change?

```rust
fn main() {
panic!("line 2 column 0");
}
```

@rfcbot concern 0

**Comment by sfackler** [comment]
vim seems to start at 1 for lines and columns.

**Comment by est31** [comment]
Compiler errors in both Rust and gcc start at 1 for columns as well. If you have code like:

```rust
fn main() {
 panic!("line 2 column 0");
}
```

Rust will say the column is 1 and if you do kate src/main.rs:linenum:1 then it jumps to the space, not the panic.

I'd say this is a off-by-one error and a bug in the actual code that emits the panic column info. I'd also say that it's a bug with the `column!()` macro but what to do about that one might be something for a later discussion.

**Comment by est31** [comment]
Note that this code gives you the correct column number:

```rust
fn main() {
let v = [1,2,3];
v[7];
}
```

I guess I'll just add one to the column value.

**Comment by est31** [comment]
Not sure how to add a regression test for this. Any ideas?

**Comment by sfackler** [comment]
You should be able to make an rpass test that sets a panic handle that looks at the column.

**Comment by est31** [comment]
@sfackler good idea, will do

**Comment by dtolnay** [comment]
@rfcbot resolved 0

**Comment by dtolnay** [comment]
We discussed this with the libs team and we would be interested in fixing the `column!()` macro as well. I filed #46868 to follow up in a separate PR.

**Comment by est31** [comment]
I've marked this as WIP until the column issue is resolved.

**Comment by est31** [comment]
I have added a test to verify that the returned column is 1 based.

@aturon @sfackler could you have a look and r+ the proposed FCP? Thanks!

**Comment by alexcrichton** [comment]
ping @aturon/@sfackler, y'all may be interested in this!

**Comment by sfackler** [comment]
LGTM

**Comment by est31** [comment]
@sfackler can you do rfcbot reviewed? Or can someone check it on @sfackler 's behalf?

**Comment by carols10cents** [comment]
ping @aturon this is waiting for your ticky box [here](https://github.com/rust-lang/rust/pull/46762#issuecomment-352452766)!

File: src/libstd/panicking.rs [hunk 1]
```diff
@@ -329,7 +328,7 @@ impl<'a> Location<'a> {
     ///
     /// panic!("Normal panic");
     /// ```
-    #[unstable(feature = "panic_col", reason = "recently added", issue = "42939")]
+    #[stable(feature = "panic_col", since = "1.24")]
```
Comment (goodmanjonathan) [inline src/libstd/panicking.rs]: Shouldn't this be since = "1.25" now?

Comment (est31) [reply src/libstd/panicking.rs]: Indeed.

**Comment by est31** [comment]
re-r? @alexcrichton

**Pull request merged**